File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    n, m, r = len(arr1), len(arr1[0]), len(arr2[0])
    answer = []

    for row in arr1:
        L = []
        for i in range(r):
            val = 0
            for j in range(m):
                val += row[j] * arr2[j][i]
                # print(i, j, val)
            L.append(val)
        answer.append(L)
                
            
            
    
    return answer

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int newR = arr1.length;
        int newC = arr2[0].length;
        int[][] answer = new int[newR][newC];
        
        for(int i = 0; i < arr1.length; i++){
            for(int j = 0; j < arr2[0].length; j++){
                for(int k = 0; k < arr2.length; k++){
                    // System.out.println("i : " + i + ", j : " + j + ", k : " + k);
                    answer[i][j] += arr1[i][k] * arr2[k][j];
                }                
            }
        }
        return answer;
    }
}

// 1 4     3 3     15 15
// 3 2     3 3     15 15               
// 4 1             15 15
// 3*2     2*2      3*2    

// arr1[0][0] * arr2[0][0] + arr1[0][1] * arr2[1][0], arr1[0][0] * arr2[0][1] + arr1[0][1] * arr2[1][1]
// arr1[i][j] * arr2[i][j] + arr1[i][k] * arr2[k][j], arr1[i][j] * arr2[i][j] + arr1[i][k] * arr2[k][j]